docs/docs-beta/docs/reference/workspace-yaml.md

```diff
@@ -0,0 +1,101 @@
+# Workspace File Reference
+
+:::info
+    This reference is only applicable to Dagster OSS, for Dagster Cloud see [the Dagster Cloud Code Locations guide](/dagster-plus/deployment/code-locations)
+:::
+
+The `workspace.yaml` file is used to configure code locations in Dagster. It tells Dagster where to find your code and how to load it.
+
+## File Structure
+
+The `workspace.yaml` file uses the following structure:
+
+~~~yaml
+load_from:
+  - <loading_method>:
+      <configuration_options>
+~~~
+
+Where `<loading_method>` can be one of:
+- `python_file`
+- `python_module`
+- `grpc_server`
+
+## Loading Methods
+
+### python_file
+
+Loads a code location from a Python file.
+
+**Options:**
+- `relative_path`: Path to the Python file, relative to the `workspace.yaml` location.
+- `attribute` (optional): Name of a specific repository or function returning a `RepositoryDefinition`.
+- `working_directory` (optional): Custom working directory for relative imports.
+- `executable_path` (optional): Path to a specific Python executable.
+- `location_name` (optional): Custom name for the code location.
+
+**Example:**
+~~~yaml
+load_from:
+  - python_file:
+      relative_path: hello_world_repository.py
+      attribute: hello_world_repository
+      working_directory: my_working_directory/
+      executable_path: venvs/path/to/python
+      location_name: my_location
+~~~
+
+### python_module
+
+Loads a code location from a Python module.
+
+**Options:**
+- `module_name`: Name of the Python module to load.
+- Other options are the same as `python_file`.
+
+**Example:**
+~~~yaml
+load_from:
+  - python_module:
+      module_name: hello_world_module.definitions
```

Comment:
Should `python_module` come first if that is the most common approach?

Also, should we call out which approach is most common, or why you would choose one over the other?

Reply:
It's a good question. I am hesitant to include that in a reference, but perhaps it's worth linking to a guide which walks through the most common approach?